client/src/common.rs

```diff
@@ -204,11 +206,18 @@ pub mod block_hash {
             .display_order(order)
     }
 
-    pub(crate) fn get(matches: &ArgMatches) -> Option<BlockHash> {
+    pub(crate) fn get(matches: &ArgMatches) -> Option<BlockIdentifier> {
         matches.value_of(ARG_NAME).map(|hex_str| {
-            let hash = Digest::from_hex(hex_str)
-                .unwrap_or_else(|error| panic!("cannot parse as a block hash: {}", error));
-            BlockHash::new(hash)
+            if hex_str.len() == 64 {
```

Comment:
Let's us `Digest::LENGTH * 2` here rather than `64` to minimise usage of magic numbers.

Reply:
Sounds good, that was an oversight on my part

Reply:
Done